src/config.rs

```diff
@@ -0,0 +1,399 @@
+/// container runtime.
+///
+/// Format defined [here](https://github.com/opencontainers/image-spec/blob/v1.0/config.md)
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, Default)]
+pub struct ConfigFile {
+    /// An combined date and time at which the image was created,
+    /// formatted as defined by
+    /// [RFC 3339, section 5.6](https://tools.ietf.org/html/rfc3339#section-5.6)
+    #[serde(
+        skip_serializing_if = "Option::is_none",
+        deserialize_with = "primitive_date_time_from_str",
+        serialize_with = "serialize_time"
+    )]
+    pub created: Option<DateTime<Utc>>,
+
+    /// Gives the name and/or email address of the person or entity
+    /// which created and is responsible for maintaining the image.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub author: Option<String>,
+
+    /// The CPU architecture which the binaries in this image are
+    /// built to run on.
+    pub architecture: Architecture,
+
+    /// The name of the operating system which the image is built to run on.
+    /// Validated values are listed in [Go Language document for GOOS](https://golang.org/doc/install/source#environment)
+    pub os: Os,
+
+    /// The execution parameters which SHOULD be used as a base when running a container using the image.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub config: Option<Config>,
+
+    /// The rootfs key references the layer content addresses used by the image.
+    pub rootfs: Rootfs,
+
+    /// Describes the history of each layer.
+    #[serde(skip_serializing_if = "Vec::is_empty")]
+    pub history: Vec<History>,
+}
+
+/// The execution parameters which SHOULD be used as a base when running a container using the image.
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, Default)]
+#[serde(rename_all = "PascalCase")]
+pub struct Config {
+    /// The username or UID which is a platform-specific structure
+    /// that allows specific control over which user the process run as. This acts as a default value to use when the value is
+    /// not specified when creating a container. For Linux based
+    /// systems, all of the following are valid: `user`, `uid`,
+    /// `user:group`, `uid:gid`, `uid:group`, `user:gid`. If `group`/`gid` is
+    /// not specified, the default group and supplementary groups
+    /// of the given `user`/`uid` in `/etc/passwd` from the container are
+    /// applied.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub user: Option<String>,
+
+    /// A set of ports to expose from a container running this
+    /// image. Its keys can be in the format of: `port/tcp`, `port/udp`,
+    /// `port` with the default protocol being `tcp` if not specified.
+    /// These values act as defaults and are merged with any
+    /// specified when creating a container.
+    #[serde(skip_serializing_if = "HashMap::is_empty")]
+    pub exposed_ports: HashMap<String, serde_json::Value>,
```

Comment:
The type it deserializes to in Go, which unfortunately influences what the spec looks like, is `map[string]struct{}`. So essentially it is acting as a set. So we have two options here:

1. The easier option (which I think I recommend for this PR): Make this a `HashMap<String, ()>` rather than a JSON value
2. The more Rust-y option (but requires more effort): Make this a `HashSet<String>` and then write a custom serializer/deserializer function for it

Reply:
I tried but it seems that `()` would be serialized into a `null` rather than a `{}`. Thus can not meet our needs. Another way I thought was to define an empty structure with no member fields. How does it like?